crates/autopilot/src/database/order_events.rs

```diff
@@ -16,21 +17,65 @@ impl super::Postgres {
     /// elaborate error handling is not necessary because this is just
     /// debugging information.
     pub async fn store_order_events(&self, events: &[(OrderUid, OrderEventLabel)]) {
-        if let Err(err) = store_order_events(self, events, Utc::now()).await {
+        if let Err(err) = store_all_order_events(self, events, Utc::now()).await {
             tracing::warn!(?err, "failed to insert order events");
         }
     }
 
+    /// Inserts the given events with the current timestamp into the DB for the
+    /// market orders only.
+    pub async fn store_market_order_events(&self, events: &[(OrderUid, OrderEventLabel)]) {
+        if let Err(err) = store_market_order_events(self, events, Utc::now()).await {
+            tracing::warn!(?err, "failed to insert market order events");
+        }
+    }
+
     /// Deletes events before the provided timestamp.
     pub async fn delete_order_events_before(&self, timestamp: DateTime<Utc>) -> Result<u64, Error> {
         order_events::delete_order_events_before(&self.0, timestamp).await
     }
 }
 
-async fn store_order_events(
+#[async_trait]
+pub trait OrderEventPersister {
```

Comment:
Just tried to avoid code duplication since both methods use almost the same logic, the only difference is the insert async operation. Couldn't make it work specifying the lifetime. `async_trait` only helped.